### PR TITLE
Avoid internal api in streaming packages

### PR DESCRIPTION
**Pull request by dims**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
`streaming` package is used in cri-o, containerd and probably other runtimes as well. So we should avoid exposing internal apis as the first step to splitting this package out somewhere public down the line

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by feiskyer — APPROVED** [review]
this is cool

/lgtm

File: pkg/kubelet/server/portforward/.import-restrictions [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+	"Rules": [
+	  {
+		"SelectorRegexp": "k8s[.]io/kubernetes",
+		"AllowedPrefixes": [
+		],
+		"ForbiddenPrefixes": [
+		]
```
Comment (liggitt) [inline pkg/kubelet/server/portforward/.import-restrictions]: what do restrictions with empty allowed and forbidden lists do?

**Comment by liggitt** [comment]
package changes lgtm, just wasn't sure whether import restrictions with empty forbidden prefix lists were effective

File: pkg/kubelet/server/portforward/.import-restrictions (hunk 1 above)
Comment (dims) [reply pkg/kubelet/server/portforward/.import-restrictions]: empty AllowedPrefixes implies there is nothing that is allowed. technically i can drop ForbiddenPrefixes from here as it has no effect. will do that shortly

**Review by mattjmcnaughton — COMMENTED** [review]
Very cool stuff - thanks for your thoughts here @dims :)

**Pull request merged**